perllib/FixMyStreet/Cobrand/Bromley.pm

```diff
@@ -374,16 +391,42 @@ sub open311_contact_meta_override {
     @$meta = grep { !$ignore{$_->{code}} } @$meta;
 }
 
+=head2 _has_report_been_sent_to_echo
+
+Assumes a report has been sent to Echo if the external ID is a GUID.
+
+=cut
+
+sub _has_report_been_sent_to_echo {
+    my ($self, $report) = @_;
+    my $guid_regex = qr/^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;
+    return $report->external_id && $report->external_id =~ /$guid_regex/;
+}
+
 =head2 should_skip_sending_update
 
 Do not send updates to the backend if they were made by a staff user and
 don't have any text (public or private).
 
+Also, if an update is on a closed echo-backed report, skip it and instead
+set the report to be resent under the 'Investigation Required' category, since we
+can't update closed echo events.
+
 =cut
 
 sub should_skip_sending_update {
     my ($self, $update) = @_;
 
+    my $report = $update->problem;
+    if ($report->is_closed && $self->_has_report_been_sent_to_echo($report)) {
+        $report->set_extra_metadata('open311_category_override' => 'Investigation Required');
```

Comment:
I think in my phase2 work I called this category "Referred to Veolia Streets". The Echo to Bromley one is called "Referred to LB Bromley Streets".

Reply:
I'll rename it as part of my cominbing with my phase2 commit.